Source/FieldSolver/SpectralSolver/SpectralAlgorithms/AvgGalileanAlgorithm.cpp

```diff
@@ -0,0 +1,371 @@
+        // Extract arrays for the coefficients
+        Array4<Real> C = C_coef[mfi].array();
+        Array4<Real> S_ck = S_ck_coef[mfi].array();
+        Array4<Real> C1 = C1_coef[mfi].array();
+        Array4<Real> S1 = S1_coef[mfi].array();
+        Array4<Real> C3 = C3_coef[mfi].array();
+        Array4<Real> S3 = S3_coef[mfi].array();
+
+        Array4<Complex> Psi1 = Psi1_coef[mfi].array();
+        Array4<Complex> Psi2 = Psi2_coef[mfi].array();
+        Array4<Complex> Psi3 = Psi3_coef[mfi].array();
+        Array4<Complex> X1 = X1_coef[mfi].array();
+        Array4<Complex> X2 = X2_coef[mfi].array();
+        Array4<Complex> X3 = X3_coef[mfi].array();
+        Array4<Complex> X4 = X4_coef[mfi].array();
+        Array4<Complex> Theta2 = Theta2_coef[mfi].array();
+        Array4<Complex> A1 = A1_coef[mfi].array();
+        Array4<Complex> A2 = A2_coef[mfi].array();
+
+        Array4<Complex> CRhoold = Rhoold_coef[mfi].array();
+        Array4<Complex> CRhonew = Rhonew_coef[mfi].array();
+        Array4<Complex> Jcoef   = Jcoef_coef[mfi].array();
+        // Extract reals (for portability on GPU)
+
+        Real vx = v_galilean[0];
+        Real vy = v_galilean[1];
+        Real vz = v_galilean[2];
+
+        // Loop over indices within one box
+        ParallelFor(bx,
+        [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept
+        {
+            // Calculate norm of vector
+            const Real k_norm = std::sqrt(
+                std::pow(modified_kx[i], 2) +
+#if (AMREX_SPACEDIM==3)
+                std::pow(modified_ky[j], 2) +
+                std::pow(modified_kz[k], 2));
+#else
+                std::pow(modified_kz[j], 2));
+#endif
+
+            // Calculate coefficients
+            constexpr Real c = PhysConst::c;
+            constexpr Real c2 = PhysConst::c*PhysConst::c;
+            constexpr Real ep0 = PhysConst::ep0;
+            const Complex I{0.,1.};
+            if (k_norm != 0){
+
+                C(i,j,k) = std::cos(c*k_norm*dt);
+                S_ck(i,j,k) = std::sin(c*k_norm*dt)/(c*k_norm);
+
+                C1(i,j,k) = std::cos(0.5_rt*c*k_norm*dt);
+                S1(i,j,k) = std::sin(0.5_rt*c*k_norm*dt);
+                C3(i,j,k) = std::cos(1.5_rt*c*k_norm*dt);
+                S3(i,j,k) = std::sin(1.5_rt*c*k_norm*dt);
+
+                // Calculate dot product with galilean velocity
+                const Real kv = modified_kx[i]*vx +
+#if (AMREX_SPACEDIM==3)
+                                 modified_ky[j]*vy +
+                                 modified_kz[k]*vz;
+#else
+                                 modified_kz[j]*vz;
+#endif
+
+                const Real nu = kv/(k_norm*c);
+                const Complex theta = amrex::exp( 0.5_rt*I*kv*dt );
+                const Complex theta_star = amrex::exp( -0.5_rt*I*kv*dt );
+                const Complex e_theta = amrex::exp( I*c*k_norm*dt );
+
+                Theta2(i,j,k) = theta*theta;
+
+                if ( (nu != 1.) && (nu != 0) ) {
```

Comment:
Here I think it would be better to add the suffix `_rt` for single precision runs:
```suggestion
                if ( (nu != 1._rt) && (nu != 0._rt) ) {
```